src/metrics/gossip_metrics.go

```diff
@@ -211,24 +217,28 @@ func (c *GossipMetrics) FillMetrics(ep track.ExtendedPeerstore) {
 			if len(peerMetrics.NodeId) == 0 {
 				//fmt.Println("NodeID empty", peerMetrics.NodeId, "Adding NodeId:", peerData.NodeID.String())
 				peerMetrics.NodeId = peerData.NodeID.String()
+				newMetaFlag = true
+				msg.NodeId = peerData.NodeID.String()
 			}
-
 			if len(peerMetrics.ClientType) == 0 || peerMetrics.ClientType == "Unknown" {
 				//fmt.Println("ClientType empty", peerMetrics.ClientType, "Adding ClientType:", peerData.UserAgent)
 				peerMetrics.ClientType = peerData.UserAgent
+				newMetaFlag = true
+				msg.ClientType = peerData.UserAgent
 			}
-
 			if len(peerMetrics.Pubkey) == 0 {
 				//fmt.Println("Pubkey empty", peerMetrics.Pubkey, "Adding Pubkey:", peerData.Pubkey)
 				peerMetrics.Pubkey = peerData.Pubkey
+				newMetaFlag = true
+				msg.PubKey = peerData.Pubkey
 			}
-
 			if len(peerMetrics.Addrs) == 0 || peerMetrics.Addrs == "/ip4/127.0.0.1/0000" || peerMetrics.Addrs == "/ip4/127.0.0.1/9000" {
 				address := GetFullAddress(peerData.Addrs)
 				//fmt.Println("Addrs empty", peerMetrics.Addrs, "Adding Addrs:", address)
 				peerMetrics.Addrs = address
+				newMetaFlag = true
+				msg.Address = address
 			}
-
 			if len(peerMetrics.Country) == 0 || peerMetrics.Country == "Unknown" {
```

Comment:
I'm a bit confused with most of the checks in here. For example, here you are checking if `Country` is `Unknown` but how is that different from just `""`? I mean, you are the one who inits it as `Unknown` in `NewPeerMetrics`, so why not just init it as `""` and checking  just `if peerMetrics == ""`

Reply:
I guess that this something draged from previous versions. `Unknown`, however now serves a "type" of country for the analysis. This loop as the entire `src/metrics` package will be updated in the following weeks.